### PR TITLE
Update to use the `ff 0.13` dependency stack.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

File: .cargo/config.offline [hunk 1]
```diff
@@ -4,5 +4,20 @@ linker = "aarch64-linux-gnu-gcc"
 [source.crates-io]
 replace-with = "vendored-sources"
 
+[source."https://github.com/zcash/halo2.git"]
+git = "https://github.com/zcash/halo2.git"
+rev = "642924d614305d882cc122739c59144109f4bd3f"
```
Comment (daira) [inline .cargo/config.offline:9]: I have not checked these hashes.

File: Cargo.lock [hunk 2]
```diff
@@ -225,8 +225,19 @@ version = "0.7.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "a3c196a77437e7cc2fb515ce413a6401291578b5afc8ecb29a3c7ab957f05941"
 dependencies = [
- "ff",
- "group",
+ "ff 0.12.1",
```
Comment (daira) [inline Cargo.lock:228]: We should update bls12_381 as well.

File: Cargo.lock [hunk 3]
```diff
@@ -1587,7 +1660,7 @@ dependencies = [
  "blake2b_simd",
  "byteorder",
  "digest 0.9.0",
- "jubjub",
+ "jubjub 0.9.0",
```
Comment (daira) [inline Cargo.lock:1663]: This is what is keeping the older crate versions around.

File: Cargo.lock [hunk 4]
```diff
@@ -2322,11 +2394,19 @@ dependencies = [
  "nonempty",
 ]
 
+[[package]]
+name = "zcash_encoding"
+version = "0.2.0"
+source = "git+https://github.com/zcash/librustzcash.git?rev=b77c952583456834bf44f1a5b41412392feab210#b77c952583456834bf44f1a5b41412392feab210"
```
Comment (daira) [inline Cargo.lock:2400]: Hmm, what's bringing two versions of `zcash_encoding` in?

File: Cargo.lock (hunk 3 above)
Comment (nuttycom) [reply Cargo.lock:1663]: This is coming in through `zcash_proofs` via `redjubjub`.

**Review by daira — CHANGES_REQUESTED** [review]
Please update `redjubjub`, to avoid still depending on the old versions of `bls12_382`, `ff`, and `group`.

**Comment by nuttycom** [comment]
> Please update `redjubjub`, to avoid still depending on the old versions of `bls12_382`, `ff`, and `group`.

`redjubjub` doesn't have a release that depends upon the updated `reddsa`.

**Comment by daira** [comment]
Also there are some unvetted dependencies.

**Comment by daira** [comment]
Filed https://github.com/ZcashFoundation/redjubjub/issues/160

File: Cargo.lock (hunk 3 above)
Comment (nuttycom) [reply Cargo.lock]: Fixed by upgrading to the redjubjub patch dependency post https://github.com/ZcashFoundation/redjubjub/pull/158. Depends on https://github.com/zcash/librustzcash/pull/801.

File: Cargo.lock [hunk 5]
```diff
@@ -707,18 +704,24 @@ dependencies = [
...
 
+[[package]]
+name = "halo2_legacy_pdqsort"
+version = "0.1.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "47716fe1ae67969c5e0b2ef826f32db8c3be72be325e1aa3c1951d06b5575ec5"
+
 [[package]]
 name = "halo2_proofs"
 version = "0.2.0"
```
Comment (daira) [inline Cargo.lock:715]: Why not 0.3.0?

Comment (daira) [reply Cargo.lock:715]: Oh, because that's blocked on releasing orchard 0.4.0.

**Review by daira — DISMISSED** [review]
utACK modulo `cargo vet` failures.

**Comment by nuttycom** [comment]
The `cargo vet` issues will be addressed later, when we remove all the patch versions prior to release.

File: Cargo.lock (hunk 5 above)
Comment (nuttycom) [reply Cargo.lock:715]: Actually, you're correct, we can now update these; at the time I initially made these changes the PR updating `orchard` had not yet been merged.

Comment (nuttycom) [reply Cargo.lock]: Fixed. Depends upon https://github.com/zcash/librustzcash/pull/803

**Review by daira — APPROVED** [review]
utACK, but I cannot review the last two commits because I wrote them.

**Review by nuttycom — COMMENTED** [review]
utACK https://github.com/zcash/zcash/pull/6500/commits/ff49b812dacd5a3794966a40eab0b28f555be072 and https://github.com/zcash/zcash/pull/6500/commits/d55d12d6e2962f6300335224568f011a882410c6.

**Comment by nuttycom** [comment]
Additional `cargo vet` failures will be fixed before release.

**Pull request merged**